src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -99,10 +102,10 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
 
                 String highlighterType = field.fieldOptions().highlighterType();
                 if (highlighterType == null) {
-                    boolean useFastVectorHighlighter = fieldMapper.fieldType().storeTermVectors() && fieldMapper.fieldType().storeTermVectorOffsets() && fieldMapper.fieldType().storeTermVectorPositions();
+                    boolean useFastVectorHighlighter = highlighters.get("fvh").canHighlight(fieldMapper);
                     if (useFastVectorHighlighter) {
                         highlighterType = "fvh";
-                    } else if (fieldMapper.fieldType().indexOptions() == IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS) {
+                    } else if (highlighters.get("postings").canHighlight(fieldMapper)) {
```

Comment:
alright, that makes sense, apart from the feature for plugins, it might still be nicer to have a sorted list and go through the available highlighters,  call canHighlight and highlight using the first one that returns true? I mean rather than getting them by name one by one? Do you like this idea?